### ISSUE TITLE
ColorPicker A11y Pass

### ISSUE THREAD

**Issue by awalia13** [OfficeDev/office-ui-fabric-react#8463]
Tracking all of the issues brought up with the 3/25 accessibility pass on ColorPicker control
Details on individual bugs can be found on [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B21986988-6108-426e-b79f-214f796cc784%7D&action=default&uid=%7B21986988-6108-426E-B79F-214F796CC784%7D&ListItemId=1898&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- ~~[ ] Bug1: [Accessibility][Programmatic access ]: Label should be defined for the sliders available for changing the colors.~~ *Doesn't block essential functionality (see below); will be fixed if we do feature #8012.*
- ~~[ ] Bug2: [Accessibility][Usable]: In scan mode backward navigation focus is going on selected color panel but screen reader is providing details about table. And forward navigation focus does not go to the selected colors quare panel, sliders, and the color picker.~~ *can't reproduce/not a bug (see below)*

**Comment by awalia13** [comment]
@micahgodbolt  - We found two more issues in ColorPicker but these issues were already logged as per below. Please suggest if we need to log below bugs also in "ColorPicker A11y Pass #8463"
https://github.com/OfficeDev/office-ui-fabric-react/pull/8429
https://github.com/OfficeDev/office-ui-fabric-react/issues/8012

**Comment by ecraig12345** [comment]
#8429 is a completed pull request, not an issue--meaning it's already been fixed.

For #8012 about slider keyboard accessibility, here's what I said on that bug:

> Even without this, the ColorPicker is still somewhat accessible (the user can type in the text fields) but that's probably not ideal. The main functional difference is that the sliders allow specific adjustment in terms of of hue, saturation, and value, whereas the text fields only allow RGBA (and hex) adjustment.

Not sure who decides how essential this is to fix?

**Comment by dzearing** [comment]
@ecraig12345 bug1 seems straightforward, correct me if I'm wrong. Bug2 sounds like it could be a Narrator issue? Is this something we need to escalate?

**Comment by ecraig12345** [comment]
Thinking about it more, fixing bug 1 may only make sense if we fix #8012--correct me if I'm wrong, but I'm not sure how useful it is to read the sliders if the user can't interact with them. Otherwise we should just make the sliders hidden from the screen reader (aria-hidden and/or role=presentation?). 

I also question how important it is to fix #8012, at least for screen reader users (sighted power users might appreciate fine-grained keyboard adjustment with visual feedback). The text fields by themselves allow selecting a full range of possible colors; the sliders and rectangle just provide a different perspective. And even if we do make them keyboard accessible, the values may not be very meaningful to users--for example, the hue slider has range 0-359, but the average person won't know which color a hue value maps to. Saturation and value numbers (the rectangle) aren't all that meaningful either. 

For bug 2, I don't know enough about scan mode to evaluate.

**Comment by ecraig12345** [comment]
Bug2 had 2 parts, neither of which really requires any action.
- "Forward navigation focus does not go to the selected colors quare panel, sliders, and the color picker" -- This is by design, since those elements aren't keyboard-accessible
- "In scan mode backward navigation focus is going on selected color panel but screen reader is providing details about table." -- I can't reproduce this. The only info read is the table with the text fields.

To ensure that screen reader users don't somehow encounter the slider/rectangle elements and get confused, I'm going to make a PR adding `aria-hidden` to those elements, then close out this issue. (Reiterating, the full range of colors can be entered using the text fields, so while it would be nice to have keyboard-accessible sliders/rectangle, it's not a high priority, especially given the amount of work required.)

**Comment by ecraig12345** [comment]
Possibly going back on the aria-hidden thing now and adding full keyboard support instead...

**Comment by ecraig12345** [comment]
Closing in favor of #8012.
